src/explore-education-statistics-common/src/modules/table-tool/utils/applyTableHeadersOrder.ts

```diff
@@ -0,0 +1,439 @@
+    return true;
+  }
+  const anyGroupsMovedOutOfAxis = previousOrder.find(group =>
+    updatedOtherAxisOrder.includes(group),
+  );
+  const anyGroupsMovedIntoAxis = updatedOrder.find(group =>
+    previousOtherAxisOrder.includes(group),
+  );
+
+  if (anyGroupsMovedOutOfAxis || anyGroupsMovedIntoAxis) {
+    return true;
+  }
+  return false;
+};
+
+const getAdditionalGroups = (
+  otherAxisReorderedGroups: Filter[][],
+  reorderedGroups: Filter[][],
+  updatedGroups: Filter[][],
+) => {
+  return updatedGroups.filter(group => {
+    const matchingGroupInAxis = getMatchingGroup(reorderedGroups, group);
+    const matchingGroupInOtherAxis = getMatchingGroup(
+      otherAxisReorderedGroups,
+      group,
+    );
+    return !matchingGroupInAxis && !matchingGroupInOtherAxis;
+  });
+};
+
+/**
+ * Filter the reordered groups by ones that exist in either axis of the updated groups.
+ * If the group items were reordered, or were reordered in the original saved version, then maintain that order, if not use the default order.
+ */
+const filterAndOrderReorderedGroups = ({
+  allOriginalGroups,
+  reorderedGroups,
+  updatedGroups,
+  updatedGroupsOnOtherAxis,
+}: {
+  allOriginalGroups: Filter[][];
+  reorderedGroups: Filter[][];
+  updatedGroups: Filter[][];
+  updatedGroupsOnOtherAxis: Filter[][];
+}): Filter[][] => {
+  return reorderedGroups.reduce<Filter[][]>((acc, group) => {
+    const matchingUpdatedGroup = getMatchingGroup(updatedGroups, group);
+    const matchingOriginalGroup = getMatchingGroup(allOriginalGroups, group);
+    const groupWasReordered = matchingUpdatedGroup
+      ? isReorderedGroup(group, matchingUpdatedGroup)
+      : false;
+
+    if (matchingUpdatedGroup) {
+      const originalWasReordered = matchingOriginalGroup
+        ? isReorderedGroup(matchingOriginalGroup, matchingUpdatedGroup)
+        : false;
+      acc.push(
+        groupWasReordered || originalWasReordered
+          ? orderItems(group, matchingUpdatedGroup)
+          : matchingUpdatedGroup,
+      );
+      return acc;
+    }
+
+    const matchingGroupInOtherAxis = getMatchingGroup(
+      updatedGroupsOnOtherAxis,
+      group,
+    );
+
+    if (matchingGroupInOtherAxis) {
+      const groupWasReorderedInOtherAxis = isReorderedGroup(
+        group,
+        matchingGroupInOtherAxis,
+      );
+      const originalWasReorderedInOtherAxis = matchingOriginalGroup
+        ? isReorderedGroup(matchingOriginalGroup, matchingGroupInOtherAxis)
+        : false;
+
+      acc.push(
+        groupWasReorderedInOtherAxis || originalWasReorderedInOtherAxis
+          ? orderItems(group, matchingGroupInOtherAxis)
+          : matchingGroupInOtherAxis,
+      );
+    }
+
+    return acc;
+  }, []);
+};
+
+/**
+ * Filter the original groups by ones that exist in either axis of the updated groups.
+ * If the group items were reordered then maintain that order, if not use the default order.
+ */
+const filterAndOrderOriginalGroups = ({
```

Comment:
Similar to the above, maybe `filterAndOrderOriginalGroupItems` would make more sense?

Reply:
This has been removed now.